soc/xtensa/esp32s2/Kconfig.defconfig

```diff
@@ -46,4 +49,19 @@ config MP_NUM_CPUS
 config MINIMAL_LIBC_OPTIMIZE_STRING_FOR_SIZE
 	default n
 
+# Indicates if a console UART is used
+config ESP_CONSOLE_UART
+	bool
+	default $(dt_chosen_enabled,$(DT_CHOSEN_Z_CONSOLE)) && \
+	$(dt_chosen_has_compat,$(DT_CHOSEN_Z_CONSOLE),$(DT_COMPAT_ESPRESSIF_ESP32_UART))
+
+# During system clock initialization this config is used to avoid data lost
+# during APB frequency switching. Due to as ESP32 ROM bug, UART2 is not
+# supported for console output, hence, this case is not handled here.
+config ESP_CONSOLE_UART_NUM
+	int
+	default -1 if !ESP_CONSOLE_UART
+	default 1 if !$(dt_nodelabel_enabled,"uart0")
+	default 0
```

Comment:
same, also, why is `ESP_CONSOLE_UART` needed? seems to be an auxiliary option?

Reply:
Ditto.

Reply:
@gmarull, we need to find a way to get the UARTx number properly. Espressif's uart calls usually depends on this value (0, 1, and 2). Do you have a suggestion on how to link/get this number properly? Last time I tried adding this info into UART's driver as `peripheral_id` instance, it was not approved. Any recommendations?

Reply:
@gmarull Any opinion regarding this? This is the blocking comment right now.

Reply:
UART number will be handled in another PR.